src/aal/aal_cheri_mips.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#define SNMALLOC_VA_BITS_64
+
+extern "C"
+{
+#include <machine/cheri.h>
```

Comment:
This is a FreeBSD-specific header, but this is in platform-independent code.  Please can you provide the implementation of the CHERI cycle counter access function (which, I believe, is BERI-specific, not dependent on CHERI?)?